src/rp2_common/hardware_i2c/i2c.c

```diff
@@ -84,6 +84,10 @@ uint i2c_set_baudrate(i2c_inst_t *i2c, uint baudrate) {
     i2c->hw->fs_scl_hcnt = hcnt;
     i2c->hw->fs_scl_lcnt = lcnt;
     i2c->hw->fs_spklen = lcnt < 16 ? 1 : lcnt / 16;
+    // Set hold time of SDA during transmit to 2 for TCS34725 color sensor
```

Comment:
I dunno enough about I2C to comment on the validity of this PR, but the comment here seems a bit misleading: it implies that this hold time is set _only_ for the "TCS34725 color sensor", whereas this PR actually changes it for all I2C devices.

Reply:
Yes, I agree, the comment is a bit misleading. Would the following be better?

The IC_SDA_TX_HOLD field in the IC_SDA_HOLD register has a default value of 1 which is known to be too low for the TCS34725 color sensor so set IC_SDA_TX_HOLD to 2.

Reply:
Like I said earlier, I don't really know much about I2C, and I've not (yet) pored through the datasheet in enough detail to know what changing the `SDA_TX_HOLD` does. But with that disclaimer out of the way, it feels a "bit blunt" to change it for **all** devices when there's only _one_ device that's causing a problem? Perhaps it would be "cleaner" to leave the defaults as they are, but then provide an accessor-function for changing the hold times, which any driver-code for the TCS34725 could then use? :shrug:

Reply:
The TCS34725 does work properly on other microcontrollers with their I2C hardware without special settings. So it may be that the default for `IC_SDA_TX_HOLD` is a bit low. It would be interesting to look at the HAL-level I2C drivers for other chips that use the licensed Synopsys DW_apb_i2c peripheral that the RP2040 uses.

As for the comment, it might be phrased as: "The TCS34725 and possibly other sensors require ..."

Reply:
> it feels a "bit blunt" to change it for all devices when there's only one device that's causing a problem?

Well, I've only used three I2C devices on a Pico so far and the TCS34725 is one of them. I don't know if there are other devices causing the same problem. The exact same TCS34725 breakout board that I'm using on the Pico functions flawlessly on a Raspberry Pi 4 from user space without the need to do anything special. It would be great if nothing special needed to be done to use the TCS34725 with the pico-sdk.

> Perhaps it would be "cleaner" to leave the defaults as they are, but then provide an accessor-function for changing the hold times, which any driver-code for the TCS34725 could then use?

Introducing a new function for changing hold times would certainly be less of a risk. On the other hand, it would also introduce new complexity. SDA hold times are really low level details and not exactly easy to understand. I think it's good that the pico-sdk hides the details of SDA hold times from users. I think the current pico-sdk API for I2C is really good. You just need to call ```i2c_init(i2c_inst_t *i2c, uint baudrate)``` and you're ready to go.

Reply:
The comment has been changed. Hopefully it's more appropriate now.